### PR TITLE
Guard isSupported check in preconditioners to Trilinos >=13.0.0

### PR DESCRIPTION
**Pull request by jpthiele**
This will fix the build problems in #17312

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]

**Comment by jpthiele** [comment]
What it is still missing and I'm getting Test pickup errors is having one output file for `.with_trilinos>=13` and `.with_trilinos<13`
I get the error that a target with the same name exists.
I'll keep looking into it but if anyone knows that would be easier.

File: include/deal.II/lac/trilinos_tpetra_precondition.templates.h [hunk 1]
```diff
@@ -260,12 +260,15 @@ namespace LinearAlgebra
...
-      bool             supported =
+
+#    if DEAL_II_TRILINOS_VERSION_GTE(13, 0, 0)
+      bool supported =
         factory.isSupported<TpetraTypes::MatrixType<Number, MemorySpace>>(
           preconditioner_type);
       AssertThrow(supported,
                   ExcTrilinosIpack2PreconditionerUnsupported(
                     preconditioner_type));
+#    endif
```
Comment (bangerth) [inline include/deal.II/lac/trilinos_tpetra_precondition.templates.h:271]: Just move the initializer of the variable into the `AssertThrow` -- there is no need to have this variable named.

Comment (jpthiele) [reply include/deal.II/lac/trilinos_tpetra_precondition.templates.h:271]: Fails with 'macro "AssertThrow" passed 3 arguments, but takes just 2.

Comment (masterleinad) [reply include/deal.II/lac/trilinos_tpetra_precondition.templates.h:271]: You probably just need an extra pair of brackets around the Trilinos macro.

Comment (jpthiele) [reply include/deal.II/lac/trilinos_tpetra_precondition.templates.h]: Indeed, that fixed it...

**Review by bangerth — APPROVED** [review]

File: include/deal.II/lac/trilinos_tpetra_precondition.templates.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/lac/trilinos_tpetra_precondition.templates.h]: Yes, the preprocessor does not understand C++ and does not know that `<...>` are parentheses. Admittedly, it's not easy even for the compiler to know when they are parentheses, and when they are less-than and greater-than signs :-)

**Pull request merged**